src/components/diagrams/singleLineDiagram/single-line-diagram-content.tsx

```diff
@@ -52,40 +51,69 @@ import { mergeSx } from '../../utils/functions';
 import { useOneBusShortcircuitAnalysisLoader } from '../use-one-bus-shortcircuit-analysis-loader';
 import { DynamicSimulationEventDialog } from '../../dialogs/dynamicsimulation/event/dynamic-simulation-event-dialog';
 import { setComputationStarting, setComputingStatus } from '../../../redux/actions';
+import { AppState } from 'redux/reducer';
+
+type EquipmentMenuState = {
+    position: [number, number];
+    equipmentId: string | null;
+    equipmentType: EQUIPMENT_TYPES | null;
+    svgId: string | null;
+    display: boolean | null;
```

Comment:
i didn't realy thinked about it , because the code in javascript was using null to initialize the values so i keep it.
i changed it to use only boolean